### PR TITLE
Add istio-workspace 0.3 tasks

### PR DESCRIPTION
**Pull request by alien-ike**
# Changes

* Add Task ike-create release 0.3
* Add Task ike-delete release 0.3
* Add Task ike-session-url release 0.3

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by chmouel** [comment]
@bartoszmajsak hey, what is those multiple PR versions about ?

**Comment by bartoszmajsak** [comment]
> @bartoszmajsak hey, what is those multiple PR versions of istio workspaces about ?

We release stuff often and as part of the release process we update tekton tasks and operator hub. But all the PRs here are not merged for whatever reason :)

**Comment by bartoszmajsak** [comment]
I closed older ones and we will fix new linter findings here. Is there anything else you are wondering about?

**Comment by vdemeester** [comment]
Any reason to start at `0.3` ? 🤔

**Comment by bartoszmajsak** [comment]
> Any reason to start at `0.3` ?

There were PRs for `0.1` and `0.2` hanging for a while (see above), but I closed them in favor of this one. I don't think there's a point in using older versions anyway.

File: task/ike-create/0.3/ike-create.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      script: |
+        test ! -z "$(params.route)" && ROUTE="--route $(params.route)"
+        export ROUTE
+
+        if ! STATE=$(ike create --session $(params.session) $ROUTE --deployment $(params.target) --namespace $(params.namespace) --image $(params.image) --json); then
+          error=$?
+          echo $STATE
+          exit $error
+        fi
+        echo "$STATE" | grep \"hosts\": | cut -d \" -f 4 | tr -d '\n' > /tekton/results/url
```
Comment (alien-ike) [inline task/ike-create/0.3/ike-create.yaml:48]:
```suggestion
        test ! -z "$(params.route)" && ROUTE="--route $(params.route)"
        export ROUTE

        if ! STATE=$(ike create --session "$(params.session)" "$ROUTE" --deployment "$(params.target)" --namespace "$(params.namespace)" --image "$(params.image)" --json); then
          error=$?
          echo "$STATE"
          exit $error
        fi
        echo "$STATE" | grep \"hosts\": | cut -d \" -f 4 | tr -d '\n' > /tekton/results/url
```

File: task/ike-delete/0.3/ike-delete.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      type: string
+      description: What deployment to target
+    - name: session
+      type: string
+      description: What to call the session
+  steps:
+    - name: ike
+      image: quay.io/maistra/istio-workspace:v0.3.0
+      script: |
+        ike delete --session $(params.session) --deployment $(params.target) --namespace $(params.namespace)
```
Comment (alien-ike) [inline task/ike-delete/0.3/ike-delete.yaml:30]:
```suggestion
        ike delete --session "$(params.session)" --deployment "$(params.target)" --namespace "$(params.namespace)"
        
```

File: task/ike-session-url/0.3/ike-session-url.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  steps:
+    - name: oc
+      image: quay.io/openshift/origin-cli:4.9
+      script: |
+        if ! STATE=$(oc get session $(params.session) --namespace $(params.namespace) -o json); then
+          error=$?
+          echo $STATE
+          exit $error
+        fi
+        echo "$STATE" | grep \"hosts\": | cut -d \" -f 4 | uniq | tr -d '\n' > /tekton/results/url
```
Comment (alien-ike) [inline task/ike-session-url/0.3/ike-session-url.yaml:35]:
```suggestion
        if ! STATE=$(oc get session "$(params.session)" --namespace "$(params.namespace)" -o json); then
          error=$?
          echo "$STATE"
          exit $error
        fi
        echo "$STATE" | grep \"hosts\": | cut -d \" -f 4 | uniq | tr -d '\n' > /tekton/results/url
        
```

**Comment by bartoszmajsak** [comment]
@chmouel it seems the latest changes based on linter "fixes" are breaking up sth in our tekton e2e tests :) Hold on with the merge, I'll come back with the update.

**Comment by bartoszmajsak** [comment]
@chmouel should be good now :)

File: task/ike-create/0.3/ike-create.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: ike-create
+  labels:
+    app.kubernetes.io/version: "0.3"
```
Comment (vinamra28) [inline task/ike-create/0.3/ike-create.yaml:6]: should we add this as `0.1` version since there are no previous versions of this task?

Comment (bartoszmajsak) [reply task/ike-create/0.3/ike-create.yaml:6]: We had PRs for `0.1` and `0.2` but hence the linter has been updated in the meantime I thought I will close older ones as there's no point of using them anyway and only fix the latest one. See https://github.com/tektoncd/catalog/pull/726#issuecomment-844816786.

Setting it back to `0.1` will make our release process a bit clumsy, as we align the version of tasks (vX.Y) with our releases of the tool the tasks is using (vX.Y.Z), so it's always matching. Is it a big deal for you to start from `0.3`?

**Comment by bartoszmajsak** [comment]
Great to see some action on this PR ;)

**Pull request closed without merging**